index.js

```diff
@@ -425,4 +425,19 @@
     return JSON.parse(s);
   });
 
+  // Regex  //////////////////////////////////////////////////////////////////
+
+  //. match :: RegEx -> String -> Maybe [*]
+  //. Takes a regular expression and tests it against the supplied string
+  //. returning a Nothing if there is no match or a Just containing an
+  //. array of the the matched results.
+  //.
+  //. ```javascript
+  //. > S.match(/abcd/, 'abcdefg')
+  //. Just(['abcd'])
+  //. ```
+  S.match = R.curry(function(regex, s) {
+    return R.compose(S.toMaybe, R.match(regex))(s);
+  });
```

Comment:
Aha! It's nice to see those tests catch something. :)

My intuition suggested that since `R.match` is a curried binary function, composing it with `S.toMaybe` would produce a curried binary function. `R.compose`, though, doesn't know that `R.match` is curried.

We could curry the resulting function ourselves:

```javascript
S.match = R.curry(R.compose(S.toMaybe, R.match));
```

This should count as one vote in favour of ramda/ramda#736.